### ISSUE TITLE
Random masks in fn.readers.coco pixelwise_masks=True

### ISSUE THREAD

**Issue by JakobDexl** [NVIDIA/DALI#4855]
### Version

1.23.0

### Describe the bug.

Some segmentation masks produced by pixel_wise=True in fn.readers.coco (following the official tutorials) return incorrect values. I would expect them to be in a range between 0 and 91 (n classes of coco + background). This probably happens when there are no masks available for the image. Tested with two different downloads of coco on a DGX server a100 and on a workstation with rtx3060. 

### Minimum reproducible example

```shell
from nvidia.dali import pipeline_def
import nvidia.dali.fn as fn
import nvidia.dali.types as types
from nvidia.dali.plugin.pytorch import DALIGenericIterator, LastBatchPolicy
import pytorch_lightning as L
from tqdm import tqdm 

FILE_ROOT = '../COCO/val2017'
ANNOTATIONS_FILE = '../COCO/annotations/instances_val2017.json'
BATCH_SIZE = 16
SIZE = 384
NUM_THREADS = 8

@pipeline_def
def GetCocoPipeline(file_root, annotations_file, size=384, device='gpu', shard_id=0, num_shards=1):  
    inputs, bboxes, labels, masks = fn.readers.coco(file_root=file_root,
                                                    annotations_file=annotations_file,
                                                    pixelwise_masks=True,  # Load segmentation pixelwise mask data
                                                    shard_id=shard_id,
                                                    num_shards=num_shards,
                                                    random_shuffle=True,
                                                    name="Reader")

    images = fn.decoders.image(inputs,
                               device='mixed' if device == 'gpu' else 'cpu') 

    return images, masks


class DALICOCO(L.LightningDataModule):
    def __init__(self, file_root=FILE_ROOT, annotations_file=ANNOTATIONS_FILE,
                 size=SIZE, batch_size=BATCH_SIZE, num_threads=NUM_THREADS):
        self.file_root = file_root
        self.annotations_file = annotations_file
        self.size = size
        self.batch_size = batch_size
        self.num_threads = num_threads
        super().__init__()

    def setup(self, stage=None, cputestmode=False):
        if cputestmode:
            device_id = 0
            shard_id = 0
            num_shards = 1
        else:
            device_id = self.trainer.local_rank
            shard_id = self.trainer.global_rank
            num_shards = self.trainer.world_size

        coco_pipeline = GetCocoPipeline(self.file_root,
                                        self.annotations_file,
                                        self.size,
                                        batch_size=self.batch_size,
                                        device='gpu',
                                        device_id=device_id,
                                        shard_id=shard_id,
                                        num_shards=num_shards,
                                        num_threads=self.num_threads)


        self.train_loader = DALIGenericIterator([coco_pipeline],
                                                ['data', 'mask'],
                                                reader_name='Reader',
                                                last_batch_policy=LastBatchPolicy.PARTIAL)
    def train_dataloader(self):
        return self.train_loader


ll = []
loader = DALICOCO(batch_size=1)
loader.setup(cputestmode=True)
tritr = loader.train_dataloader()
for el in tqdm(tritr):
    tmp = el[0]["mask"].to("cpu")
    if tmp.min() < 0 or tmp.max() > 81:
        print("tm", tmp.min(), tmp.max())
        ll.append((el[0]["mask"].to("cpu"), el[0]["data"].to("cpu")))
```

### Relevant log output

```shell
2%|▉                                      | 122/5000 [00:00<00:08, 608.37it/s]

tm tensor(0, dtype=torch.int32) tensor(1883416992, dtype=torch.int32)

 29%|███████████                           | 1454/5000 [00:02<00:05, 604.90it/s]

tm tensor(-2147449334, dtype=torch.int32) tensor(2147400654, dtype=torch.int32)

 47%|█████████████████▊                    | 2351/5000 [00:03<00:03, 671.28it/s]

tm tensor(0, dtype=torch.int32) tensor(36570, dtype=torch.int32)
tm tensor(0, dtype=torch.int32) tensor(1949607376, dtype=torch.int32)

 51%|███████████████████▍                  | 2559/5000 [00:04<00:03, 632.66it/s]

tm tensor(0, dtype=torch.int32) tensor(9997697, dtype=torch.int32)

 73%|███████████████████████████▌          | 3634/5000 [00:05<00:02, 656.77it/s]

tm tensor(0, dtype=torch.int32) tensor(52431, dtype=torch.int32)

 80%|██████████████████████████████▌       | 4024/5000 [00:06<00:01, 585.43it/s]

tm tensor(0, dtype=torch.int32) tensor(2082422768, dtype=torch.int32)

 85%|████████████████████████████████▍     | 4264/5000 [00:06<00:01, 574.81it/s]

tm tensor(0, dtype=torch.int32) tensor(8783217, dtype=torch.int32)

 90%|██████████████████████████████████▏   | 4500/5000 [00:07<00:00, 564.27it/s]

tm tensor(0, dtype=torch.int32) tensor(322, dtype=torch.int32)

100%|█████████████████████████████████████▉| 4991/5000 [00:08<00:00, 569.46it/s]

tm tensor(0, dtype=torch.int32) tensor(9038017, dtype=torch.int32)

100%|██████████████████████████████████████| 5000/5000 [00:08<00:00, 608.63it/s]
```

### Check for duplicates

- [X] I have searched the [open bugs/issues](https://github.com/NVIDIA/DALI/issues) and have found no duplicates for this bug report

**Comment by JanuszL** [comment]
Hi @JakobDexl,

Thank you for reporting this issue.
https://github.com/NVIDIA/DALI/pull/4856 should fix that.
Please check the nightly build that follows the merge of the mentioned PR.

**Comment by JanuszL** [comment]
Release 1.27 is out. Closing.
